types/src/transaction/transaction_v1.rs

```diff
@@ -728,38 +729,35 @@ impl PartialOrd for TransactionV1 {
 }
 
 impl ToBytes for TransactionV1 {
```

Comment:
Do to nesting of serialization fields our assumtion of making the `TransactionV1` extensible with optional fields depends on the actual nested fields having the same behavior.
Maybe it's worth adding a marker trait for every field that has this behaviour so the behaviour is not surprising and it's also easier to maintain (e.g. if someone adds a complex field to some part of the transaction body then it needs to implement the map serialization behaviour).

Reply:
No, there is no such assumption.